music21/humdrum/spineParser.py

```diff
@@ -2119,7 +2122,27 @@ def getAllOccurring(self):
         return retEvents
 
 
-def hdStringToNote(contents):
+@lru_cache(maxsize=256)
+def _noteMemos(contents):
+    dotCount = contents.count('.')
+    accountedFor = dotCount
+    dur_regex = re.search(r'(\d+)%?(\d+)?', contents)
+    if dur_regex:
+        accountedFor += len(dur_regex.group(0))
+    if 'r' in contents:
+        accountedFor += 1
+        matchedNote = None
+        step = None
+    else:
+        matchedNote = re.search('([a-gA-G]+)([n#-]*)', contents)
+        if not matchedNote:
+            raise HumdrumException(f'Could not parse {contents} for note information')
+        accountedFor += len(matchedNote.group(0))
+        step = matchedNote.group(1)[0].lower()
+    return (dur_regex, dotCount, matchedNote, step, accountedFor)
```

Comment:
don't mix underscore and camelCase in internal variable names.  Choose one or the other.